runtime/wasm/src/host_exports.rs

```diff
@@ -812,6 +813,27 @@ pub(crate) fn bytes_to_string(logger: &Logger, bytes: Vec<u8>) -> String {
     s.trim_end_matches('\u{0000}').to_string()
 }
 
+pub(crate) fn abi_encode(params: Vec<Token>) -> Result<Vec<u8>, anyhow::Error> {
+    Ok(encode(&params))
+}
+
+pub(crate) fn abi_decode(mut types: String, data: Vec<u8>) -> Result<Vec<Token>, anyhow::Error> {
+    if !(types.starts_with('(') && types.ends_with(')')) {
+        return Err(anyhow::anyhow!("'ethabi' requires types to start and finish with parentheses, eg: '(uint256,address,(bytes32,int8,address))'"));
+    }
+
+    types.pop();
+    types.remove(0);
+
+    let param_types: Vec<ParamType> = types
+        .split(',')
+        .map(Reader::read)
+        .collect::<Result<_, _>>()
+        .or_else(|e| Err(anyhow::anyhow!("Failed to read types: {}", e)))?;
+
+    decode(&param_types, &data).map_err(|e| anyhow::anyhow!("Failed to decode: {}", e))
```

Comment:
Both the `or_else` and the `map_err` look like they can be replaced by anyhow's `context` https://docs.rs/anyhow/1.0.40/anyhow/trait.Context.html#tymethod.context.

Reply:
Nice, fixed!